src/middleware/auth/logto.ts

```diff
@@ -50,6 +49,135 @@ export class LogToHelper {
     return this.allResourceWithNames
   }
 
+  public async setDefaultRoleForUser(userId: string): Promise<ILogToErrorResponse | void> {
+    const roles = await this.getRolesForUser(userId)
+    if (roles && roles.status === 200 && roles.data.length === 0) {
+        const default_role = await this.getDefaultRole()
+        if (default_role && default_role.status === 200) {
+            return await this.assignDefaultRoleForUser(userId, default_role.data.id)
+        }
+    }
+  }
+
+  private async assignDefaultRoleForUser(userId: string, roleId: string): Promise<ILogToErrorResponse | void> {
+    const uri = new URL(`/api/roles/${roleId}/users`, process.env.LOGTO_ENDPOINT);
+    try {
+        const body = {
+            userIds: [userId],
+        };
+        const response = await fetch(uri, {
+            headers: {
+                Authorization: 'Bearer ' + this.m2mToken,
+                'Content-Type': 'application/json',
+            },
+            method: 'POST',
+            body: JSON.stringify(body),
+        });
+
+        if (response.ok) {
+            
+            return {
+                status: 200,
+                error: "",
+                data: {}
+            }
+        }
+        return {
+            error: await response.text(),
+            status: response.status,
+            data: {}
+        }
+
+        
+    } catch (err) {
+        return {
+            error: `getRolesForUser ${err}`,
+            status: 500,
+            data: {}
+        }
+    }
+  }
```

Comment:
Seems unnecessarily complex to use the Roles API and then check all users, rather than going under the Users API, for a specific user, and then using the GET and POST endpoints: https://docs.logto.io/api/#tag/Users/paths/~1api~1users~1:userId~1roles/post